services/apps/premium/members_enrichment_worker/src/activities/getMembers.ts

```diff
@@ -15,31 +15,34 @@ export async function getMembers(): Promise<IMember[]> {
   try {
     rows = await svc.postgres.reader.connection().query(
       `SELECT
-          members."id",
-          members."displayName",
-          members."attributes",
-          members."emails",
-          members."contributions",
-          members."score",
-          members."reach",
-          members."tenantId",
-          jsonb_object_agg(mi.platform, mi.username) as username
-        FROM members
-        INNER JOIN tenants ON tenants.id = members."tenantId"
-        INNER JOIN "memberIdentities" mi ON mi."memberId" = members.id
-        WHERE tenants.plan IN ('Scale', 'Enterprise')
-        AND (
-          members."lastEnriched" < NOW() - INTERVAL '90 days'
-          OR members."lastEnriched" IS NULL
-        )
-        AND (
-          mi.platform = 'github'
-          OR array_length(members.emails, 1) > 0
-        )
-        AND tenants."deletedAt" IS NULL
-        AND members."deletedAt" IS NULL
-        GROUP BY members.id
-        LIMIT 50;`,
+        members."id",
+        members."displayName",
+        members."attributes",
+        members."emails",
+        members."contributions",
+        members."score",
+        members."reach",
+        members."tenantId",
+        jsonb_object_agg(mi.platform, mi.username) as username,
+        COUNT(activities."memberId") AS activity_count
+      FROM members
+      INNER JOIN tenants ON tenants.id = members."tenantId"
+      INNER JOIN "memberIdentities" mi ON mi."memberId" = members.id
+      INNER JOIN activities ON activities."memberId" = members.id
```

Comment:
I'm a bit worried this query will be heavy for members with many activities. @sausage-todd might have an opinion?

Reply:
No matter the `LIMIT 50` in the end of the query, it will take **all the members** in the database, join **all their activities**, filter through that, group, sort, and only then take the first 50.
On LF database it takes 60 seconds.
Joining with `mv_activities_cube` instead of `activities` doesn't help much.

If we will execute this query once every half an hour, I think it's fine.